### PR TITLE
Add sfx djinn set/unset, and djinn trade/give, asset for level up sfx

### PR DESCRIPTION
**Pull request by SamuelPhang**
Working towards issue #226

### CODE REVIEW

File: base/windows/djinn/DjinnListWindow.ts [hunk 1]
```diff
@@ -67,6 +67,7 @@ export class DjinnListWindow {
     public selected_djinn_index: number;
     public action_text_selected: boolean;
     public page_index: number;
+    public djinn_chosen: boolean;
```
Comment (jjppof) [inline base/windows/djinn/DjinnListWindow.ts:70]: Let's see whether those `this.grant_full_user_control()` calls are necessary. Then we can decide whether this variable is truly necessary.

**Review by jjppof — CHANGES_REQUESTED** [review]
Many thanks for the help!
Let's confirm the necessity of `this.grant_full_user_control()` calls. Please see the comments.
The sfx conditions when setting the controls are fine.

File: base/windows/djinn/DjinnListWindow.ts [hunk 2]
```diff
@@ -223,7 +230,10 @@ export class DjinnListWindow {
         else {
             if (this.action_text_selected || this.selected_djinn_index === 0) {
                 this.select_djinn(this.selected_char_index, this.sizes[this.selected_char_index] - 1, true);
-            } else this.select_djinn(this.selected_char_index, this.selected_djinn_index - 1);
+            } else {
+                this.select_djinn(this.selected_char_index, this.selected_djinn_index - 1);
+                if (!this.setting_djinn_status) this.grant_full_user_control();
```
[the remark this answers is not in the document]
Comment (SamuelPhang) [reply base/windows/djinn/DjinnListWindow.ts:235]: Every time the cursor is moved we need to update the sfx connected to the R button to make sure it plays the correct sound when setting/unsetting a djinn. So I added this call to each function that is bound to a direction pad button press. Let me know if there's a better way

[the remark this answers is not in the document]
Comment (jjppof) [reply base/windows/djinn/DjinnListWindow.ts:235]: Got it. And yeah, there's no better way, but I'll do some changes in ControlManager to support this. I'll let you know. Thanks!

**Review by jjppof — APPROVED** [review]

**Pull request merged**